src/Microsoft.ML.Core/Utilities/FuncInstanceMethodInfo1`2.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Linq.Expressions;
+using System.Reflection;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Internal.Utilities
+{
```

Comment:
Any idea why the file names of the newly added files in this directory have an `1 or `2 suffix? Is that an accident or is there a specific reason behind this naming?

Reply:
The file names match the names of the types defined in the files. When generic types are compiled, the compiler adds a `` `{arity} `` suffix to the type name.

Reply:
What does the `{arity}` represent here?

---
In reply to: [386548523](https://github.com/dotnet/machinelearning/pull/4588#discussion_r386548523) [](ancestors = 386548523)

Reply:
The number of generic type arguments.

Reply:
I'm using the term from here:
https://github.com/dotnet/roslyn/blob/6389e2519f282f7683025761189894e1c894936c/src/Compilers/Core/Portable/Symbols/INamedTypeSymbol.cs#L23-L27

I wanted to link to the language specification but it seems the specification doesn't explicitly give a name to this characteristic of generic types.

Reply:
Thanks for the additional info. Please feel free to merge. 

---
In reply to: [386566851](https://github.com/dotnet/machinelearning/pull/4588#discussion_r386566851) [](ancestors = 386566851)

Reply:
I'll run a build locally to make sure nothing changed in the interim and merge if it passes.